packages/nuxt/src/core/runtime/nitro/error.ts

```diff
@@ -2,7 +2,8 @@ import { joinURL, withQuery } from 'ufo'
 import type { NitroErrorHandler } from 'nitropack'
 import type { H3Error } from 'h3'
 import { getRequestHeaders, send, setResponseHeader, setResponseStatus } from 'h3'
-import { useNitroApp, useRuntimeConfig } from '#internal/nitro'
+import { useRuntimeConfig } from '#internal/nitro'
+import { useNitroApp } from '#internal/nitro/app'
```

Comment:
Have you tried with `import { useNitroApp } from '#imports'`? it should be safer to not introduce internal dependencies.

Reply:
This isn't introducing a dependency - we already import `useNitroApp` elsewhere from this path.

Using `#imports` in the Nuxt source code means we don't get type support.

Reply:
I would appreciate if we can try to avoid internal imports also for other places sooner  🙏🏼  (We can internally add type back I guess, from one place)

Reply:
I'm a bit confused. I removed this very import but you then said we didn't need to remove it, here:

https://github.com/nuxt/nuxt/pull/20391#discussion_r1298416478

Reply:
Anyway, I agree supporting `#imports` would be nice internally. 👍 

I'll go ahead and merge this PR as it doesn't change internal situation and we can look to move safely to `#imports` at any point after that.